scripts/staging/hmm/HMM Training.py

```diff
@@ -0,0 +1,135 @@
+import numpy as np
+import pandas as pd
+import random
+
+def add2dict(dictionary, key, value):
+    if key not in dictionary:
+        dictionary[key] = []
+    dictionary[key].append(value)
+
+def list2probabilitydict(given_list):
+    probability_dict = {}
+    given_list_length = len(given_list)
+    for item in given_list:
+        probability_dict[item] = probability_dict.get(item, 0) + 1
+    for key, value in probability_dict.items():
+        probability_dict[key] = value / given_list_length
+    return probability_dict
+
+def sample_word(dictionary):
+    key = np.random.choice(list(dictionary.keys()), 1, p = list(dictionary.values()))[0]
+    return key
+
+def generate(sentence, no_of_words):
+    sentence = sentence.split()
+    if len(sentence) == 0:
+        word0 = sample_word(initial_word)        
+        sentence.append(word0)
+
+    if len(sentence) == 1:
+        word0 = sentence[0]
+        if word0 in second_word.keys():
+            word1 = sample_word(second_word[word0])
+        else:
+            word1 = np.random.choice(list(initial_word.keys()), 1, p = list(initial_word.values()))[0]
+        sentence.append(word1)
+        no_of_words = no_of_words - 1
+    
+    if len(sentence) == 2:
+        word0 = sentence[-2]
+        word1 = sentence[-1]
+        if (word0, word1) in transitions.keys():
+            word2 = sample_word(transitions[(word0, word1)])
+        else:
+            if word0 in second_word.keys():
+                word2_tmp1 = sample_word(second_word[word0])
+                dict1 = second_word[word2_tmp1]
+                word2_tmp1 = sample_word(dict1)
+            else:
+                word2_tmp1 = np.random.choice(list(second_word.keys()), 1)
+                dict1 = second_word[word2_tmp1[0]]
+                word2_tmp1 = sample_word(dict1)
+            if word1 in second_word.keys():
+                word2_tmp2 = sample_word(second_word[word1])
+                dict2 = second_word[word2_tmp2]
+                word2_tmp2 = sample_word(dict2)
+            else:
+                word2_tmp2 = np.random.choice(list(second_word.keys()), 1)
+                dict2 = second_word[word2_tmp2[0]]
+                word2_tmp2 = sample_word(dict2)
+            word2 = word2_tmp1 if dict1[word2_tmp1] > dict2[word2_tmp2] else word2_tmp2
+        sentence.append(word2)
+        no_of_words = no_of_words - 1
+    
+    while no_of_words > 0:
```

Comment:
Fix the length of predicted words (no_of_words) to 5. It is generating errors on random inputs such as 12 or 15.